packages/settings/src/settings.ts

```diff
@@ -59,13 +50,80 @@ export class Settings {
    *
    */
   async save(plugin: string, raw: string): Promise<void> {
-    await this._storage.setItem(plugin, raw);
+    await Private._storage.setItem(plugin, raw);
   }
+}
 
-  private _storage = localforage.createInstance({
+/**
+ * A namespace for private data
+ */
+namespace Private {
+  export const _storage = localforage.createInstance({
     name: STORAGE_NAME,
     description: 'Offline Storage for Settings',
     storeName: 'settings',
     version: 1
   });
+
+  /**
+   * Get all the settings.
+   */
+  export async function getAll(): Promise<{ settings: IPlugin[] }> {
+    const settingsUrl = PageConfig.getOption('settingsUrl') ?? '/';
+    const all = (await (
+      await fetch(URLExt.join(settingsUrl, 'all.json'))
+    ).json()) as IPlugin[];
+    const settings = await Promise.all(
+      all.map(async plugin => {
+        const { id } = plugin;
+        const raw = ((await _storage.getItem(id)) as string) ?? plugin.raw;
+        return {
+          ...plugin,
+          raw,
+          settings: json5.parse(raw)
+        };
+      })
+    );
+    return { settings };
+  }
+
+  /**
+   * Get the settings for a federated extension
+   *
+   * @param id The id of the federated extension
+   */
+  export async function getFederated(id: string): Promise<IPlugin | undefined> {
+    const [ext, plugin] = id.split(':');
+    let federated: string[];
```

Comment:
oops, i think this comes back as a `{name, load, extension, style}`... will update...